snapdtool/fips_linux.go

```diff
@@ -99,19 +99,20 @@ func MaybeSetupFIPS() error {
 
 	logger.Debugf("snapd snap: %s", currentRevSnapdSnap)
 
-	exe, err := osReadlink(selfExe)
+	rootDir, exe, err := exeAndRoot()
 	if err != nil {
 		return err
 	}
 
 	logger.Debugf("self exe: %s", exe)
+	logger.Debugf("exe root dir: %q", rootDir)
 
 	// on a classic system we need to be reexecuted from the snapd snap for
 	// the FIPS setup to be relevant, but on core we are not reexeced but
 	// running directly from the mount of the snapd snap under
 	// /usr/lib/snapd, yet we still want to set up the right environment
 	if release.OnClassic {
-		if !strings.HasPrefix(exe, currentRevSnapdSnap+"/") {
+		if !strings.HasPrefix(rootDir, dirs.SnapMountDir) {
```

Comment:
wouldn't at this point rootDir and  currentRevSnapdSnap be the same? do we need to compute  currentRevSnapdSnap?

Reply:
Ah yes, good point. Updated.